sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -825,7 +825,7 @@ public ReadFiles<T> withDatumReaderFactory(AvroSource.DatumReaderFactory<T> fact
 
     @Override
     public PCollection<T> expand(PCollection<ReadableFile> input) {
-      checkNotNull(getSchema(), "schema");
+      checkArgumentNotNull(getSchema(), "schema");
```

Comment:
Here also `checkStateNotNull`. And if you feel like making a better error message, that is nice too.